### ISSUE TITLE
Navigator Error

### ISSUE THREAD

**Issue by NelsonUpenn** [ContinuumIO/anaconda-issues#3287]
## Main error
Application <b>spyder</b> launch may have produced errors.
## Traceback
```
Traceback (most recent call last):
  File "/Users/pcn/anaconda/bin/spyder", line 6, in <module>
    sys.exit(spyder.app.start.main())
  File "/Users/pcn/anaconda/lib/python3.6/site-packages/spyder/app/start.py", line 103, in main
    from spyder.app import mainwindow
  File "/Users/pcn/anaconda/lib/python3.6/site-packages/spyder/app/mainwindow.py", line 165, in <module>
    from spyder.utils.environ import WinUserEnvDialog
  File "/Users/pcn/anaconda/lib/python3.6/site-packages/spyder/utils/environ.py", line 20, in <module>
    from spyder.widgets.variableexplorer.collectionseditor import CollectionsEditor
  File "/Users/pcn/anaconda/lib/python3.6/site-packages/spyder/widgets/variableexplorer/collectionseditor.py", line 46, in <module>
    from spyder.widgets.variableexplorer.importwizard import ImportWizard
  File "/Users/pcn/anaconda/lib/python3.6/site-packages/spyder/widgets/variableexplorer/importwizard.py", line 26, in <module>
    import pandas as pd
  File "/Users/pcn/anaconda/lib/python3.6/site-packages/pandas/__init__.py", line 42, in <module>
    from pandas.core.api import *
  File "/Users/pcn/anaconda/lib/python3.6/site-packages/pandas/core/api.py", line 10, in <module>
    from pandas.core.groupby import Grouper
  File "/Users/pcn/anaconda/lib/python3.6/site-packages/pandas/core/groupby.py", line 49, in <module>
    from pandas.core.frame import DataFrame
  File "/Users/pcn/anaconda/lib/python3.6/site-packages/pandas/core/frame.py", line 74, in <module>
    from pandas.core.series import Series
  File "/Users/pcn/anaconda/lib/python3.6/site-packages/pandas/core/series.py", line 56, in <module>
    import pandas.core.strings as strings
AttributeError: module 'pandas' has no attribute 'core'

```
## System information
```
python: 3.6.1
language: en
os: Darwin;16.6.0;Darwin Kernel Version 16.6.0: Fri Apr 14 16:13:31 PDT 2017; root:xnu-3789.60.24~4/RELEASE_X86_64;x86_64;i386
version: 1.6.2
platform: osx-64
qt: 5.6.2
pyqt: 5.6.0
conda: 4.3.21
```
## More background: I previously had the same problem with an older MacOS10.9 as well.

**Comment by goanpeca** [comment]
@ccordoba12 ?

**Comment by NelsonUpenn** [comment]
Update: I followed a tip from Kerry Watson at Continuum and added two lines to my ~/.bash_profile, namely
```
export LC_ALL=en_US.UTF-8
export LANG=en_US.UTF-8
```
Sadly this had no effect. However, it may be relevant for me to show here the contents of ~/bash_profile
```
# MacPorts Installer addition on 2015-02-12_at_06:51:26: adding an appropriate \
PATH variable for use with MacPorts.                                            
export PATH="/opt/local/bin:/opt/local/sbin:$PATH"
# Finished adapting your PATH environment variable for use with MacPorts.       
# added by Anaconda3 4.3.1 installer                                            
export PATH="/Users/pcn/anaconda/bin:$PATH"

```

**Comment by NelsonUpenn** [comment]
And here is the weird resolution. Summary: I'm now back in business, but have followup questions.

I discovered I had an empty folder called ~/matplotlibrc. I deleted it, and now Spyder launches! (Apparently I created this long ago.) This folder was bad in two ways: The correct thing is a **file* named matplotlibrc contained in a **folder** named ~/.matplotlib/ .

Although I am in business, two disturbing behaviors persist:

1. When I launch from the Unix command line, I succeed but get mysterious messages in the terminal:
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
[warn] kq_init: detected broken kqueue; not using.: Undefined error: 0

I ignored all of that. (When I launch from Anaconda Navigator, I also succeed but these same messages appear in a popup.)

2. I also got a mysterious popup saying "gcc requires the command line developer tools. Would you like to install them?" 
I said "not now", and everything appears to work fine.

So my remaining question is: Is there something I should do to avoid the two nonfatal behaviors (1--2)?

**Comment by goanpeca** [comment]
@NelsonUpenn  from @ccordoba12 

Is there something I should do to avoid the two nonfatal behaviors (1--2)?
For point 1. the solution is adding this line to ~/.bashrc:

export EVENT_NOKQUEUE=1
This is also fixed in the latest Navigator version, if you launch Spyder through it.

I haven't seen point 2. before, so I can't recommend a solution for it.

Also, your developers may want to think about whether the system is acting too fragile in the face of what is admittedly an obscure goof (my bogus ~/matplotlibrc/ folder ).
I added several modifications to Spyder to make it more robust when importing third-party modules:

spyder-ide/spyder#4765

This work will be part of Spyder 3.2, to be released, as I said before, at the end of this week.

**Comment by ccordoba12** [comment]
This was fixed in spyder-ide/spyder#4765.
